File: 1752-CheckifArrayIsSortedandRotated/1752-CheckifArrayIsSortedandRotated.cpp
// Last updated: 23/5/2026, 5:16:04 pm
1class Solution {
2public:
3    bool check(vector<int>& nums) {
4        int count = 0, n = nums.size();
5        for (int i = 0; i < n; i++) {
6            if (nums[i] > nums[(i + 1) % n])
7                count++;
8        }
9        return count <= 1;
10    }
11};

File: 1752-CheckIfArrayIsSortedAndRotated/1752-CheckIfArrayIsSortedAndRotated.py
# Last updated: 9/5/2026, 12:27:50 AM
class Solution:
    def check(self, nums: List[int]) -> bool:
        drops = 0
        l = len(nums)
        for i in range(0, l):
            if nums[i] > nums[(i+1)%l]:
                drops+=1
            if drops > 1:
                return False
        return True